mlir-sycl/lib/Dialect/SYCL/IR/SYCLDialect.cpp

```diff
@@ -205,8 +205,11 @@ SYCLOpAsmInterface::getAlias(mlir::Type Type, llvm::raw_ostream &OS) const {
         return AliasResult::OverridableAlias;
       })
       .Case<mlir::sycl::VecType>([&](auto Ty) {
-        OS << "sycl_" << decltype(Ty)::getMnemonic() << "_" << Ty.getDataType()
-           << "_" << Ty.getNumElements() << "_";
+        OS << "sycl_" << decltype(Ty)::getMnemonic() << "_";
+        mlir::Type DataTy = Ty.getDataType();
+        if (getAlias(DataTy, OS) == AliasResult::NoAlias)
+          OS << DataTy;
+        OS << "_" << Ty.getNumElements() << "_";
```

Comment:
Needed to have nice-looking aliases, e.g., `!sycl_vec_sycl_half_16_`

Reply:
`DataTy` is not part of the alias if `getAlias(DataTy, OS) != AliasResult::NoAlias`? Should we use `getAlias(DataTy, OS) == AliasResult::NoAlias`?

Reply:
`getAlias(DataTy, OS)` will be:
- `FinalAlias` or `OverridableAlias` if the input type (`DataTy`) has an alias and such alias has been passed to `OS`, e.g., will be part of the output;
- `NoAlias` if `DataTy` has no alias, so nothing has been passed to `OS`.

As we only support a handful of types here, we know what will happen for each of them:
- `i[1|8|16|32|64]` -> `NoAlias`, we can safely write `iN` here anyway;
- `f[32|64]` -> `NoAlias`, but same;
- `!sycl.half<(...)>` -> `FinalAlias` (`sycl_half`), this will be written to `OS` by the call to `getAlias`.